src/TLuaInterpreter.cpp

```diff
@@ -10042,102 +10062,125 @@ int TLuaInterpreter::setGridMode(lua_State* L)
 int TLuaInterpreter::setFgColor(lua_State* L)
 {
-    int s = 1;
+    int s = 0;
     int n = lua_gettop(L);
-    string a1;
+    QString windowName;
     int luaRed;
     int luaGreen;
     int luaBlue;
     if (n > 3) {
-        if (lua_isstring(L, s)) {
-            a1 = lua_tostring(L, s);
-            s++;
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "setFgColor: bad argument #%d type (window name as string expected, got %s!)", s, luaL_typename(L, s));
+            return lua_error(L);
+        } else {
+            windowName = QString::fromUtf8(lua_tostring(L, s));
         }
     }
-    if (!lua_isnumber(L, s)) {
-        lua_pushstring(L, "setFgColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setFgColor: bad argument #%d type (red color as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
         luaRed = lua_tointeger(L, s);
-        s++;
+        if (luaRed < 0 || luaRed >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's red component value %d is outside of the valid range (0 to 255)", luaRed);
+            return lua_error(L);
+        }
     }
 
-    if (!lua_isnumber(L, s)) {
-        lua_pushstring(L, "setFgColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setFgColor: bad argument #%d type (green color as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
         luaGreen = lua_tointeger(L, s);
-        s++;
+        if (luaGreen< 0 || luaGreen >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's green component value %d is outside of the valid range (0 to 255)", luaGreen);
+            return lua_error(L);
+        }
     }
 
-    if (!lua_isnumber(L, s)) {
-        lua_pushstring(L, "setFgColor: wrong argument type");
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setFgColor: bad argument #%d type (blue color as number expected, got %s!)", s, luaL_typename(L, s));
         lua_error(L);
         return 1;
     } else {
         luaBlue = lua_tointeger(L, s);
+        if (luaBlue < 0 || luaBlue >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's blue component value %d is outside of the valid range (0 to 255)", luaBlue);
+            return lua_error(L);
+        }
     }
 
-    QString _name(a1.c_str());
     Host& host = getHostFromLua(L);
-    if (n < 4) {
-        host.mpConsole->setFgColor(luaRed, luaGreen, luaBlue);
-    } else {
-        mudlet::self()->setFgColor(&host, _name, luaRed, luaGreen, luaBlue);
+    if (windowName.isEmpty() || windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) == 0) {
+        windowName = "main";
     }
+
+    mudlet::self()->setFgColor(&host, windowName, luaRed, luaGreen, luaBlue);
```

Comment:
Sorry but `setFgColor()` is ***also*** not working now for the **main** console - you need to revert this block - so that the above line goes back to being:
```C+
    if (n < 4) {
        host.mpConsole->setFgColor(luaRed, luaGreen, luaBlue);
    } else {
        mudlet::self()->setFgColor(&host, _name, luaRed, luaGreen, luaBlue);
    }
```

Reply:
Okay, Vadim has reminded me we ought to also detect `main` (or the empty string) being used as to specify the main console case, so I'd recommend now:
```C++
    if (n < 4 || windowName.isEmpty() || windowName.compare(QLatin1String("main")) == 0 ) {
        host.mpConsole->setBgColor(luaRed, luaGreen, luaBlue);
    } else {
        mudlet::self()->setBgColor(&host, windowName, luaRed, luaGreen, luaBlue);
    }
```

Reply:
applied.